homeassistant/components/skybell/binary_sensor.py

```diff
@@ -12,90 +10,78 @@
-        key="device:sensor:button",
+BINARY_SENSOR_TYPES: tuple[BinarySensorEntityDescription, ...] = (
+    BinarySensorEntityDescription(
+        key="button",
         name="Button",
         device_class=BinarySensorDeviceClass.OCCUPANCY,
     ),
-    "motion": BinarySensorEntityDescription(
-        key="device:sensor:motion",
+    BinarySensorEntityDescription(
+        key="motion",
         name="Motion",
         device_class=BinarySensorDeviceClass.MOTION,
     ),
-}
-
+)
 
+# Deprecated in Home Assistant 2022.6
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Optional(
-            CONF_ENTITY_NAMESPACE, default=DEFAULT_ENTITY_NAMESPACE
-        ): cv.string,
+        vol.Optional(CONF_ENTITY_NAMESPACE, default=DOMAIN): cv.string,
         vol.Required(CONF_MONITORED_CONDITIONS, default=[]): vol.All(
             cv.ensure_list, [vol.In(BINARY_SENSOR_TYPES)]
         ),
     }
 )
 
 
-def setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
-    """Set up the platform for a Skybell device."""
-    skybell = hass.data[SKYBELL_DOMAIN]
-
-    binary_sensors = [
-        SkybellBinarySensor(device, BINARY_SENSOR_TYPES[sensor_type])
-        for device in skybell.get_devices()
-        for sensor_type in config[CONF_MONITORED_CONDITIONS]
-    ]
+    """Set up Skybell switch."""
+    async_add_entities(
+        SkybellBinarySensor(coordinator, sensor)
+        for sensor in BINARY_SENSOR_TYPES
+        for coordinator in hass.data[DOMAIN][entry.entry_id]
+    )
 
-    add_entities(binary_sensors, True)
 
-
-class SkybellBinarySensor(SkybellDevice, BinarySensorEntity):
+class SkybellBinarySensor(SkybellEntity, BinarySensorEntity):
     """A binary sensor implementation for Skybell devices."""
 
     def __init__(
         self,
-        device,
+        coordinator: SkybellDataUpdateCoordinator,
         description: BinarySensorEntityDescription,
-    ):
+    ) -> None:
         """Initialize a binary sensor for a Skybell device."""
-        super().__init__(device)
+        super().__init__(coordinator)
         self.entity_description = description
         self._attr_name = f"{self._device.name} {description.name}"
-        self._event: dict[Any, Any] = {}
+        self._event: dict[str, str] = {}
+        self._attr_unique_id = f"{self._device.device_id}_{description.key}"
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, str | int | tuple[str, str]]:
         """Return the state attributes."""
         attrs = super().extra_state_attributes
-
-        attrs["event_date"] = self._event.get("createdAt")
-
+        if event := self._event.get(CONST.CREATED_AT):
+            attrs["event_date"] = event
```

Comment:
Existing, but for a future PR:

https://developers.home-assistant.io/docs/core/entity?_highlight=entity#generic-properties

Entities that generate a significant amount of state changes can quickly increase the size of the database when the extra_state_attributes also change frequently. Minimize the number of extra_state_attributes for these entities by removing non-critical attributes or creating additional sensor entities.